auth/authenticator.go

```diff
@@ -30,9 +30,9 @@ type authenticator struct {
 	keySecret  string
 }
 
-// New returns a new instance of an authenticator
+// NewAuthenticator returns a new instance of an authenticator
 // that conforms to the Authenticator interface
-func New(instanceID, keyID, keySecret string) Authenticator {
+func NewAuthenticator(instanceID, keyID, keySecret string) Authenticator {
```

Comment:
Just `New` was fine, now there's more stutter.  Most people don't distinguish AuthN and AuthZ, and those that do will figure it out.